src/justenoughharfbuzz.c

```diff
@@ -254,6 +256,12 @@ int shape (lua_State *L) {
       lua_pushstring(L, "depth");
       lua_pushnumber(L, -tHeight - height);
       lua_settable(L, -3);
+      lua_pushstring(L, "x_bearing");
+      lua_pushnumber(L, x_bearing);
+      lua_settable(L, -3);
+      lua_pushstring(L, "glyphWidth");
+      lua_pushnumber(L, glyphWidth);
+      lua_settable(L, -3);
```

Comment:
Yes, this will need to be preceded by `lua_checkstack(L, 4)` because you're pushing 4 more values on the stack and you need to make sure the VM is ready to allocate enough addresses. The existing check can just be bumped by 4.

I would hold off rebasing this, since it isn't a breaking change (is it?) we might even get it in a point release. Once we're sure I can probably retarget it pretty easily having done the same operation a few times recently.

Reply:
It might be _very slightly_ breaking on hbox construction logic (but so slightly that I'm unsure any existing user might be affected, unless they did something weird), hence my initial suggestion initially for having it in 0.15 rather than a 0.14.x.

FWIW, for my next official[^2] published book made with SILE[^3], soon to be announced[^1], I'll use a special SILE build with this feature (amongst other things) --> So there's no urge for rebasing etc. -- This is just a heads up / reminder as I noticed #1793 might need an additional check here indeed, not to forget in due time. (But don't hesitate letting me know if a rebase is needed or whatever.)

[^1]: Reviews and proofreading are ongoing, and we still have a chapter missing -- but the plan so far is to get the book ready in Nov. 2023... We are late by all standards, and Nov. 2023 is a best attempt, I'll be fairly busy already that month -- Hence also my (relative) silence lately ;-)

[^2]: What I mean by "official" here = ISBN-registered + (French law applies) legal deposit completed. I've the first, still need to finalize the second ('cause it needs the final page count to be fully completed, and it thus depends whether a missing chapter makes it or not).

[^3]: It's not even a secret, the (current, partially reviewed at this point) book sources are available in one of my repos ;)